jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Context.java

```diff
@@ -38,7 +38,8 @@
 public interface Context extends Attributes, Decorator, Executor
 {
     /**
-     * @return the encoded context path of this Context
+     * @return the encoded context path of this {@code Context} or
+     *         {@code null} if this is the {@link org.eclipse.jetty.server.Server.ServerContext}.
```

Comment:
I would specify generically `null` here, and specialize in `ServerContext`.
Seems to me not ideal to talk about behavior of subinterfaces in the parent interface.

Reply:
It is a bit hard to describe when a null is a valid return without mentioning the server context???

Reply:
@gregw you can specialize the javadoc in `ServerContext` and and in `ScopedContext` and here just say: "or {@code null} if the context path has not been determined yet".